core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4

```diff
@@ -3321,17 +3334,17 @@ withExpression: WITH commonTableExpression (COMMA commonTableExpression)*
     ;
 
 commonTableExpression
-    : tableName = id (L_PAREN columns += id (COMMA columns += id)* R_PAREN)? AS L_PAREN selectStatement setOperators* R_PAREN
+    : tableName = id (L_PAREN columns += id (COMMA columns += id)* R_PAREN)? AS L_PAREN ((selectStatement setOperators*) | expr) R_PAREN
     ;
 
 selectStatement
     : selectClause selectOptionalClauses limitClause?
     | selectTopClause selectOptionalClauses //TOP and LIMIT are not allowed together
+    | L_PAREN selectStatement R_PAREN
     ;
 
 setOperators
-    : (UNION ALL? | EXCEPT | MINUS_ | INTERSECT) selectStatement //EXCEPT and MINUS have same SQL meaning
-    | L_PAREN selectStatement R_PAREN
+    : (UNION ALL? | EXCEPT | MINUS_ | INTERSECT) selectStatement  //EXCEPT and MINUS have same SQL meaning
```

Comment:
If we are claning up we should probably rename tokens that have a trailing _ as they were meant to avoid symbol clashes in various generated targets (where keywords like NULL would clash with target language builtins). ANTLR does that automatically now - this is not enough to request changes though.